File: src/com/binarysushi/studio/debugger/StudioDebuggerEditorsProvider.java
package com.binarysushi.studio.debugger;

import com.intellij.lang.javascript.JavaScriptFileType;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.xdebugger.evaluation.XDebuggerEditorsProvider;
import org.jetbrains.annotations.NotNull;

public class StudioDebuggerEditorsProvider extends XDebuggerEditorsProvider {
    @NotNull
    @Override
    public FileType getFileType() {
        return JavaScriptFileType.INSTANCE;
    }
}

File: src/main/kotlin/com/binarysushi/studio/debugger/StudioDebuggerEditorsProvider.kt
package com.binarysushi.studio.debugger

import com.intellij.lang.javascript.JavaScriptFileType
import com.intellij.openapi.editor.Document
import com.intellij.openapi.editor.EditorFactory
import com.intellij.openapi.fileTypes.FileType
import com.intellij.openapi.project.Project
import com.intellij.xdebugger.XExpression
import com.intellij.xdebugger.XSourcePosition
import com.intellij.xdebugger.evaluation.EvaluationMode
import com.intellij.xdebugger.evaluation.XDebuggerEditorsProvider

class StudioDebuggerEditorsProvider : XDebuggerEditorsProvider() {
    override fun getFileType(): FileType {
        return JavaScriptFileType.INSTANCE
    }

    override fun createDocument(
        project: Project,
        expression: XExpression,
        sourcePosition: XSourcePosition?,
        mode: EvaluationMode
    ): Document {

        return EditorFactory.getInstance().createDocument(expression.expression)
    }
}
